libsnark/zk_proof_systems/plonk/srs.tcc

```diff
@@ -121,26 +237,23 @@ srs<ppT> plonk_srs_derive_from_usrs(
         secret_powers_g2.push_back(usrs.secret_powers_g2[i]);
     }
 
-    std::shared_ptr<libfqfft::evaluation_domain<libff::Fr<ppT>>> domain =
-        libfqfft::get_evaluation_domain<libff::Fr<ppT>>(circuit.num_gates);
-
     // compute 0-th Lagrange basis vector via inverse FFT
-    polynomial<libff::Fr<ppT>> u(circuit.num_gates, libff::Fr<ppT>(0));
+    polynomial<libff::Fr<ppT>> u(num_gates, libff::Fr<ppT>(0));
```

Comment:
Do you know if we can just set `polynomial<libff::Fr<ppT>> u{1};` and have `domain->iFFT` expand it to the right length?  It might not work, but if it does it would be slightly clearer.  Also, $u$ is not used anywhere else, so instead of copying it to `L_basis_zero` only to move it to the srs, we can just start with `L_basis_zero` variable.